plugins/processor/desensitize/processor_desensitize.go

```diff
@@ -134,23 +134,28 @@ func (p *ProcessorDesensitize) desensitize(val string) string {
 	}
 
 	var pos = 0
-	match, _ := p.regexBegin.FindStringMatchStartingAt(val, pos)
+	runeVal := []rune(val)
+	match, _ := p.regexBegin.FindRunesMatchStartingAt(runeVal, pos)
 	for match != nil {
 		pos = match.Index + match.Length
-		content, _ := p.regexContent.FindStringMatchStartingAt(val, pos)
+		content, _ := p.regexContent.FindRunesMatchStartingAt(runeVal, pos)
 		if content != nil {
 			if p.Method == "const" {
-				val, _ = p.regexContent.Replace(val, p.ReplaceString, pos, 1)
-				pos = content.Index + len(p.ReplaceString)
+				curPos := len(string(runeVal[:pos]))
+				val, _ = p.regexContent.Replace(val, p.ReplaceString, curPos, 1)
+				runeVal = []rune(val)
```

Comment:
This could be a quick fix. However, the code performance will be poor. We may just find the positions to be replaced in one loop and construct the desensitized string in another. This can avoid reconstruct the rune slice repeatedly.